### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by tal47**
@DxCx

### CODE REVIEW

File: dapulseBot.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    return r >= 200 and r < 300
+
+
+def getUsers():
+    users = []
+    r = requests.get("https://api.dapulse.com:443/v1/users.json?api_key=" + sys.argv[1] ) # api_key
+    if not httpOK(r.status_code):
+        return []
+    data = json.loads(r.content.decode('utf-8'))
+    for user in data:
```
Comment (DxCx) [inline dapulseBot.py:44]: users = [(user['name'], user['id']) for user in users];

File: dapulseBot.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    retUser = ""
+    task = text[:text.find("<")] 
+    user = text.split("<",1)[1][:-1]
+
+    users = getUsers()
+    if not users:
+        return "Error getting users!"
+
+    for name, ids in users:
+        if user.lower() in name.lower():
```
Comment (DxCx) [inline dapulseBot.py]: user.lower() == name.lower()

File: dapulseBot.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+            retId = ids 
+            retUser = name 
+
+    if retUser == "":
+        return "User " + user + " is wrong. fix and try again"
+
+    url = "https://api.dapulse.com:443/v1/boards/" + sys.argv[3] + "/pulses.json?api_key=" + sys.argv[1] # board_id, api_key
```
Comment (DxCx) [inline dapulseBot.py:66]: don't get them from sys.argv here, you should pass some config dictunry to the function which have `{'boardId': ...., 'apiKey': ...}`

File: dapulseBot.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    payload = { "pulse": { "name": task }, "user_id": retId }
+
+    r = requests.post(url, data=json.dumps(payload))
+    print(r)
```
Comment (DxCx) [inline dapulseBot.py:70]: can be cleaned up

File: dapulseBot.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    if not httpOK(r.status_code):
+        return "Error! Code: " + str(r.status_code)
+    return "Added task: " + task + ", Dest user: " + retUser 
+
+def main():
+    init()
+    token = sys.argv[2] #token
```
Comment (DxCx) [inline dapulseBot.py:77]: argument of main as i said above

**Review by DxCx — COMMENTED** [review]
Good Job! :) the only real issue is the usage of sys inside the code :)

**Comment by tal47** [comment]
Thanks man!

**Pull request closed without merging**